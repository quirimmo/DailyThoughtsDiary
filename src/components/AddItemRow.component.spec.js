import React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';
import AddItemRow from './AddItemRow.component';

const mockOnChangeText = jest.fn();
let component;
let instance;

describe('AddItemRow', () => {
	beforeEach(() => {
		jest.useFakeTimers();
		NavigationTestUtils.resetInternalState();
		component = renderer.create(
			<AddItemRow
				value="value"
				label="label"
				placeholder="placeholder"
				onChangeText={mockOnChangeText}
			/>
		);
		instance = component.getInstance();
	});
	afterEach(() => {
		jest.clearAllMocks();
	});

	it('should be defined', () => {
		expect(AddItemRow).toBeDefined();
	});

	it('should render the component', async () => {
		expect(component.toJSON()).toMatchSnapshot();
	});

	describe('state', () => {
		it('should init the value property', () => {
			expect(instance.state.value).toEqual('value');
		});
	});

	describe('onChangeText', () => {
		let textInput;
		beforeEach(() => {
			textInput = component.root.find(element => element.type === 'TextInput');
		});

		it('should set the value property of the state to the given value', () => {
			textInput.props.onChangeText('input-text');
			expect(instance.state.value).toEqual('input-text');
		});

		it('should call the onChangeText prop', () => {
			textInput.props.onChangeText('input-text');
			expect(mockOnChangeText).toHaveBeenCalledWith('input-text');
		});
	});
});
