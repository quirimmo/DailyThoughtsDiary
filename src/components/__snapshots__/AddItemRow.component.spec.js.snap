// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItemRow should render the component 1`] = `
<View
  style={
    Object {
      "marginBottom": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={undefined}
  >
    label
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="placeholder"
    style={
      Object {
        "fontSize": 16,
        "height": 50,
      }
    }
    value="value"
  />
</View>
`;
